### PR TITLE
Adding Code of Conduct (#590)

### PR DESCRIPTION
**Pull request by aepfli**
based on https://www.contributor-covenant.org/

Proposed commit message:

```
Adding Code of Conduct (#590 / #592) 

Adding Code of Conduct based on https://www.contributor-covenant.org/

closes: #590 
PR: #592 
```

---
**PR checklist**

The following checklist shall help the PR's author, the reviewers and maintainers to ensure the quality of this project.
It is based on our contributors guidelines, especially the ["writing code" section](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#writing-code).
It shall help to check for completion of the listed points.
If a point does not apply to the given PR's changes, the corresponding entry can be simply marked as done. 

Documentation (general)
* [ ] There is documentation (Javadoc and site documentation; added or updated)
* [ ] There is implementation information to describe _why_ a non-obvious source code / solution got implemented
* [ ] Site documentation has its own `.adoc` file in the `docs` folder, e.g. `docs/report-entries.adoc`
* [ ] Only one sentence per line (especially in `.adoc` files)
* [ ] Javadoc uses formal style, while sites documentation may use informal style

Documentation (new extension)
* [ ] The `docs/docs-nav.yml` navigation has an entry for the new extension
* [ ] The `package-info.java` contains information about the new extension

Code
* [ ] Code adheres to code style, naming conventions etc.
* [ ] Successful tests cover all changes
* [ ] There are checks which validate correct / false usage / configuration of a functionality and there are tests to verify those checks
* [ ] Tests use [AssertJ](https://assertj.github.io/doc/) or our own [PioneerAssert](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#assertions) (which are based on AssertJ)

Contributing
* [x] A prepared commit message exists
* [x] The list of contributions inside `README.md` mentions the new contribution (real name optional) 

---

I hereby agree to the terms of the [JUnit Pioneer Contributor License Agreement](https://github.com/junit-pioneer/junit-pioneer/blob/main/CONTRIBUTING.md#junit-pioneer-contributor-license-agreement).

### CODE REVIEW

**Review by Michael1993 — COMMENTED** [review]
Does not adhere to the single-line per sentence rule. :(

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+or harmful.
+
+Community leaders have the right and responsibility to remove, edit, or reject
+comments, commits, code, wiki edits, issues, and other contributions that are
+not aligned to this Code of Conduct, and will communicate reasons for moderation
+decisions when appropriate.
+
+## Scope
+
+This Code of Conduct applies within all community spaces, and also applies when
```
Comment (Michael1993) [inline CODE_OF_CONDUCT.md:53]: Should we mention that it applies to the Discord channel? 😅

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -50,11 +50,12 @@ decisions when appropriate.
...
-an individual is officially representing the community in public spaces.
-Examples of representing our community include using an official e-mail address,
-posting via an official social media account, or acting as an appointed
-representative at an online or offline event.
+This Code of Conduct applies within all community spaces (this includes but is
+not limited to the project's presence on GitHub and the nipafx Discord server),
+and also applies when an individual is officially representing the community in
+public spaces. Examples of representing our community include using an official
+e-mail address, posting via an official social media account, or acting as an
+appointed representative at an online or offline event.
```
Comment (nipafx) [inline CODE_OF_CONDUCT.md:58]: And this diff, my friends, is why you put each sentence on a single line.

File: CODE_OF_CONDUCT.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
[the remark this answers is not in the document]
Comment (nipafx) [reply CODE_OF_CONDUCT.md]: See comment further down.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (nipafx) [reply CODE_OF_CONDUCT.md]: Yes!

File: CODE_OF_CONDUCT.md [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+appointed representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to Nicolai Parlog ([contact options](https://nipafx.dev/contact))
+or the other maintainers (via Discord DMs). All complaints will be reviewed and
+investigated fairly and as promptly as circumstances allow (as explained in
+[the contribution guide](CONTRIBUTING.md#pioneer-maintainers), there's no
+expectation of availability towards maintainers).
```
Comment (nipafx) [inline CODE_OF_CONDUCT.md:67]: Are you guys ok with this? (I.e. people potentially reaching out via Discord DM)

**Review by Bukama — APPROVED** [review]

**Comment by nipafx** [comment]
@aepfli, @Michael1993 If you could formally approve this PR, that would be nice.

**Review by Michael1993 — APPROVED** [review]

**Comment by aepfli** [comment]
> @aepfli, @Michael1993 If you could formally approve this PR, that would be nice.

i think i can not approve my own PR :) but i approve with this comment

**Comment by nipafx** [comment]
Commit message:

```
Add Code of Conduct (#590 / #592) 

Add a code of conduct based on
https://www.contributor-covenant.org/ .

Closes: #590 
PR: #592 
```

**Pull request merged**